src/template.rs

```diff
@@ -119,52 +120,57 @@ pub fn substitute(
 pub fn walk_dir(
     project_dir: &PathBuf,
     template: liquid::value::Object,
+    template_config: Option<TemplateConfig>,
     pbar: ProgressBar,
 ) -> Result<(), failure::Error> {
-    fn is_dir(entry: &DirEntry) -> bool {
-        entry.file_type().is_dir()
-    }
-
-    fn is_git_metadata(entry: &DirEntry) -> bool {
-        entry
-            .path()
-            .to_str()
-            .map(|s| s.contains(".git"))
-            .unwrap_or(false)
-    }
-
     let engine = engine();
 
-    for entry in WalkDir::new(project_dir) {
-        let entry = entry?;
-        if is_dir(&entry) || is_git_metadata(&entry) {
-            continue;
+    //returning iterators is hard :/
+    match template_config {
+        Some(template_config) => {
+            for entry in create_matcher(&template_config, project_dir)? {
```

Comment:
Clippy is angry about this line and 136 on latest stable. It's a bug in clippy:  https://github.com/rust-lang/rust-clippy/issues/4133